extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -26,11 +121,12 @@ export let KeyedShoppingConfigDef;
 export function getShoppingConfig(pageElement) {
   const element = pageElement.querySelector('amp-story-shopping-config');
   return getElementConfig(element).then((config) => {
-    //TODO(#36412): Add call to validate config here.
+    for (const configItem of config['items']) {
+      validateConfig(configItem);
```

Comment:
Yes, this is the intended behavior - it sounds like you may have a better approach in mind - feel free to share if so.

Reply:
Great discussion. I think there can be a balance between what's required to display and not displaying at all.

These fields are required to display a shopping tag and be rendered in the PLP:
`productPrice`, `productTitle`, `productImages` (at least one), `productBrand`, `productId`
This can be considered essential because they are part of the core user journey.

The validation will throw errors but the components will still render if any of the following are missing or misconfigured:
`productUrl`: If invalid or missing, gray out "buy now" button.
`aggregateRating`: Do not render reviews module if missing from config or misconfigured.
`productPriceCurrency`: There is already a smart default built in, it renders the number but no currency label.

Reply:
I'm going to go ahead and make this change.
@processprocess (if you haven't already) would you mind updating the i2i with this functionality?

Reply:
In order to disable the rendering of the shopping tag and PLP when an essential field is missing, the keys for the shopping tag data must be keyed properly in order to test this properly. Therefore, I will wait until #37600 is in before attempting to block rendering.

Reply:
After a discussion with @processprocess  have decided for this particular PR to simply go with a "block" when any of the required fields are missing. Essential fields are not going to be a part of this PR, but may be a part of future PRS. This is done for two reasons:

1) To keep the scope of this PR within reason (i.e. avoid feature creep).
2) The problem of having the UX is speculative, i.e. tooling may mitigate some of the pain points brought up. The dev preview is out right now, and spending dev resources/time on a problem before it has been reviewed or tested is not ideal. In other words, this is a developer speculation problem at the current moment, not anything users have actually reported as cumbersome.

The discussion of essential fields / other UX validation partial loads (i.e. grayed-out instead of fully colored review stars, etc.) is moved to this doc for now. if you would like to discuss this point further, please do so in the doc. 

https://docs.google.com/document/d/1ohPni-Muv9bcLRPOJ5bWyTPEtp2TDMLI_lfQDsxJcW0/edit?resourcekey=0-I-D1nbEwituQJGDIGhiYtA#

We will open a new issue to address this as/if needed. Thanks.